controllers/temporalworkerprocess_controller.go

```diff
@@ -0,0 +1,243 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/alexandrevilain/temporal-operator/api/v1beta1"
+	temporaliov1beta1 "github.com/alexandrevilain/temporal-operator/api/v1beta1"
+	"github.com/alexandrevilain/temporal-operator/pkg/resource"
+	"github.com/alexandrevilain/temporal-operator/pkg/status"
+	"github.com/alexandrevilain/temporal-operator/pkg/workerprocess"
+)
+
+// TemporalWorkerProcessReconciler reconciles a TemporalWorkerProcess object
+type TemporalWorkerProcessReconciler struct {
+	client.Client
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+}
+
+//+kubebuilder:rbac:groups=temporal.io,resources=temporalworkerprocesses,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=temporal.io,resources=temporalworkerprocesses/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=temporal.io,resources=temporalworkerprocesses/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the TemporalWorkerProcess object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
+func (r *TemporalWorkerProcessReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	logger.Info("Starting reconciliation")
+
+	worker := &v1beta1.TemporalWorkerProcess{}
+	err := r.Get(ctx, req.NamespacedName, worker)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	// Check if the resource has been marked for deletion
+	if !worker.ObjectMeta.DeletionTimestamp.IsZero() {
+		logger.Info("Deleting worker process", "name", worker.Name)
+		return reconcile.Result{}, nil
+	}
+
+	// Set namespace based on ClusterRef
+	namespace := worker.Spec.ClusterRef.Namespace
+	if namespace == "" {
+		namespace = req.Namespace
+	}
+
+	namespacedName := types.NamespacedName{Namespace: namespace, Name: worker.Spec.ClusterRef.Name}
+	cluster := &v1beta1.TemporalCluster{}
+	err = r.Get(ctx, namespacedName, cluster)
+	if err != nil {
+		return r.handleError(ctx, worker, v1beta1.ReconcileErrorReason, err)
+	}
+
+	if err := r.reconcileResources(ctx, worker, cluster); err != nil {
+		logger.Error(err, "Can't reconcile resources")
+		return r.handleErrorWithRequeue(ctx, worker, v1beta1.ResourcesReconciliationFailedReason, err, 2*time.Second)
+	}
+
+	if !worker.Status.Ready {
```

Comment:
This is not needed, this should be handled in the `SetupWithManager` where you have to add:
```
Owns(&appsv1.Deployment{})
```

Using this, it will reconcile when the deployment's status will change :)

Reply:
done